src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -70,6 +70,11 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
     }
   }
 
+  KOKKOS_FUNCTION TreeTraversal(BVH const &bvh, Callback const &callback)
+      : _bvh{bvh}
+      , _callback{callback}
+  {}
```

Comment:
I don't know if the current form of `TreeTraversal` is ideal. Imho, we could move the launch loop into `BVH::query`. This way, we wouldn't need any constructors at all, and could make it a struct, or have a single constructor. 

Not confident, though.